### DISCUSSION TITLE
Is there no way to set a timeout for a route in bun.serve?

### DISCUSSION THREAD

**Discussion by BennyAlex** [oven-sh/bun#13492] [Q&A]
I currently run into an issue where I have a memory leak or a route that is running very long/for ever and will eventually break the whole server.
Looks like there is no timeout setable for the route handler in bun, is this correct? Why is that?

All I can do is to call promise.race by myself but with this approch its not possible to cancel any code that is beeing executed at the moment (and might run forever/very long) but will correctly return a 408 Error when the timeout is reached. 

Is there any way to solve this?

I found out bluebird has cancelable promises so I thought this feature would be a great addition in bun.

**Comment by rhuanbarreto** [comment]
This is what you're looking for: https://bun.com/docs/api/http#idletimeout
